projects/packages/publicize/src/social-image-generator/class-extractor.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Extractor class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize\Social_Image_Generator;
+
+/**
+ * Class for extracting SIG-related settings for a post.
+ */
+class Extractor {
+	/**
+	 * Post to get information from.
+	 *
+	 * @var int $post_id
+	 */
+	public $post_id;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param int $post_id Post to get information from.
+	 */
+	public function __construct( $post_id ) {
+		$this->post_id = $post_id;
+	}
+
+	/**
+	 * Get the SIG settings for a post.
+	 *
+	 * @return array
+	 */
+	private function get_post_settings() {
+		$social_options = get_post_meta( $this->post_id, \Automattic\Jetpack\Publicize\Publicize::POST_JETPACK_SOCIAL_OPTIONS, true );
+
+		if ( ! is_array( $social_options ) || empty( $social_options['image_generator_settings'] ) ) {
+			return array();
+		}
+
+		return $social_options['image_generator_settings'];
+	}
+
+	/**
+	 * Check if SIG is enabled for a specific post.
+	 *
+	 * @return bool
+	 */
+	public function is_enabled() {
+		$settings = $this->get_post_settings();
+
+		return ! empty( $settings['is_enabled'] );
+	}
+
+	/**
+	 * Get the text to use for the generated image.
+	 *
+	 * @return string
+	 */
+	public function get_generated_image_text() {
+		$settings = $this->get_post_settings();
+
+		if ( ! empty( $settings['custom_text'] ) ) {
+			return $settings['custom_text'];
+		}
+
+		return get_the_title( $this->post_id );
+	}
+
+	/**
+	 * Get the background image to use for the generated image.
+	 *
+	 * @return string
+	 */
+	public function get_generated_image_background_image_url() {
```

Comment:
It does need `generated`, because `get_image_url` already exists and returns the URL to the service with the token. I think this demonstrates an issue with the Extractor class (similar to what you mentioned [here](https://github.com/Automattic/jetpack/pull/29118#discussion_r1117742613)) in that it does more than just extracting information from the post.

I'm wondering if I add something like `Social_Image_Generator\Utilities::get_image_url( $post_id )` that returns the image URL for the `og:image` tag, and rename the `Extractor` class to `PostSettings` and have it just contain some helper methods to get information from the post meta? What do you think?

Reply:
Yeah that could work. Would `get_image_url` be the only method in `Utilities`? Maybe that could just be a function in the namespace. 

Or this could just be `get_background_image_url`?

Reply:
I've added a separate utilities file that has just that single method: https://github.com/Automattic/jetpack/pull/29118/commits/ad7ab30c47e8e53defd3597b676a23d6af9c27d5